runtime/pendulum/src/xcm_config.rs

```diff
@@ -92,6 +100,45 @@ impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
 	}
 }
 
+pub struct RelativeValue {
+	num: Balance,
+	denominator: Balance,
+}
+
+impl RelativeValue {
+	fn adjust_amount_by_relative_value(amount: Balance, relative_value: RelativeValue) -> Balance {
+		if relative_value.denominator == 0 {
+			// Or probably error
+			return amount
+		}
+		// Calculate the adjusted amount
+		let adjusted_amount = amount * relative_value.denominator / relative_value.num;
+		adjusted_amount
+	}
+}
+
+pub struct RelayRelativeValue;
+impl RelayRelativeValue {
+	fn get_relative_value(id: CurrencyId) -> Option<RelativeValue> {
+		match id {
+			CurrencyId::XCM(f) => match f {
+				xcm_assets::RELAY_DOT => Some(RelativeValue { num: 1, denominator: 1 }),
+				xcm_assets::ASSETHUB_USDT => Some(RelativeValue { num: 1, denominator: 4 }),
+				xcm_assets::ASSETHUB_USDC => Some(RelativeValue { num: 1, denominator: 4 }),
+				xcm_assets::EQUILIBRIUM_EQD => Some(RelativeValue { num: 1, denominator: 10 }),
+				xcm_assets::MOONBEAM_BRZ => Some(RelativeValue { num: 1, denominator: 10 }),
+				xcm_assets::POLKADEX_PDEX => Some(RelativeValue { num: 1, denominator: 2 }),
+				xcm_assets::MOONBEAM_GLMR => Some(RelativeValue { num: 1, denominator: 10 }),
+				_ => None,
+			},
+
+			CurrencyId::Native => Some(RelativeValue { num: 1, denominator: 2 }),
```

Comment:
Wouldn't it make more sense to use denominator 1 here?

Reply:
Well in this example you point out, our native would be "worth" half of what the relay chain token is. All the relative valuations are examples. If we decide to go with this solution then we should plug in the adequate fraction.

Reply:
As I point out in [this comment](https://github.com/pendulum-chain/pendulum/pull/385#pullrequestreview-1798251341), I think that the base fee is in terms of our native asset, not in the relay chain native asset.

Reply:
Oh okay, I see. Then we should have to modify our [base fee](https://github.com/pendulum-chain/pendulum/pull/385/files#diff-f2b065c9a230c62a9e0320c5dc933302af3fe4c229d63a5432eb109e7a327e71R312) to be in terms of the native token. In that case yes, the relation of course is 1:1.
Right now we are actually using the value given by `WeightToFee` as the fee in relay token, this was defined [here](https://github.com/pendulum-chain/pendulum/pull/385/files#diff-f2b065c9a230c62a9e0320c5dc933302af3fe4c229d63a5432eb109e7a327e71L263).

Reply:
I think was wrong to begin with. The gold standard (I think) is how pallet transaction payment does it and how it assigns a fee in our native asset to a particular weight.

Pallet-transaction-payment is more complicated of course and also uses an adaptive fee multiplier, which at low activity is well below 1. But assuming a multiplier of 1 and neglecting tips and length fees, then `WeightToFee` is exactly the native asset fee assigned to an extrinsic of a specific weight. More here: https://www.notion.so/satoshipay/22-05-26-Transaction-Payment-dd2304d17baa4b6d9ae0b76873069870?pvs=4